libmcount/dynamic.c

```diff
@@ -127,15 +129,25 @@ static int do_dynamic_update(struct symtabs *symtabs, char *patch_funcs)
 				stats.skipped++;
 				break;
 			case 0:
+				success = true;
 			default:
 				break;
 			}
 			stats.total++;
 		}
 
+		if (!success) {
+			nr_failed++;
+			func_name = name;
+		}
+
 		name = strtok(NULL, ";");
 	}
 
+	if (nr_failed)
+		pr_out("%s cannot be patched dynamically\n",
+		       nr_failed > 1 ? "some functions" : func_name);
+
```

Comment:
**We cannot know number of wrong function name that the user gave**,
because `stats.total`, `stats.failed` and `stats.skipped` are **0 zero value,**
even though the user gave more than one wrong function name with `-P`.

For example,
if just using `if (stats.total == 0 || (stats.failed + stats.skipped) > 0)`,
not using other way (e.g. `stats.skipped++`), a problem can occur.

### Just one given wrong function name: [OK]
Expectation: `wrong_name1 cannot be patched dynamically`
The result   : `wrong_name1 cannot be patched dynamically`
```
$ uftrace -P wrong_name1 ./hello.dyn
wrong_name1 cannot be patched dynamically
hello
foo
bar
# DURATION    TID     FUNCTION
   0.673 us [27003] | __monstartup();
   0.501 us [27003] | __cxa_atexit();
   0.770 us [27003] | puts();
   0.196 us [27003] | puts();
   0.138 us [27003] | puts();
```
### Two given wrong function names: [NG]
Expectation: `some functions cannot be patched dynamically`
The result   : `wrong_name1 cannot be patched dynamically`
```
uftrace -P wrong_name1 -P wrong_name2 ./hello.dyn
wrong_name2 cannot be patched dynamically
hello
foo
bar
# DURATION    TID     FUNCTION
   0.561 us [27030] | __monstartup();
   0.324 us [27030] | __cxa_atexit();
   0.565 us [27030] | puts();
   0.172 us [27030] | puts();
   0.090 us [27030] | puts();
```
The reason is the if statement [(dynamic.c:118~120)](https://github.com/namhyung/uftrace/blob/master/libmcount/dynamic.c#L118-L120) just skip it,
so `stats.total`, `stats.failed` and `stats.skipped` are 0 zero value
and **we cannot know number of wrong function name that the user gave**.

So, IMHO, we should other way to distinguish whether one or several wrong function names that aren't contained in symbol table.

Just show `some functions cannot be patched dynamically` ?
(whether one or several functions that aren't contained in symbol table)

Sorry for taking your time because of the warning message of lesser importance..